### DISCUSSION TITLE
Is there an option for returning the texture as square cropped from middle?

### DISCUSSION THREAD

**Discussion by gokturk-tas** [yasirkula/UnityNativeCamera#50] [Q&A] [answered]
Cropped area needs to be square as I said and if photo is portrait, crop square size should be photo(width)x photo(width)
If photo is landscape crop square size should be photo(height)x photo(height) and crop location should be in the middle.
Is there such an option or how can I make it possible?

Also thanks for the great work, I really liked your pluggin.

**Comment by gokturk-tas** [comment, marked as answer]
I came up with a solution. I used one of @Mops project script. It's link is;
[https://github.com/mopsicus/unity-crop-scale-texture/blob/master/Assets/Scripts/CropScale.cs](url) 
I editted that a little bit and here is the result.

  `private static Texture2D CropTexture(Texture2D texture)
      {
          int cropSize;
          if (texture.height > texture.width)
              cropSize = texture.width;
          else
              cropSize = texture.height;
          Rect sizes = new Rect();
          Texture2D result = new Texture2D(cropSize, cropSize);
          
          sizes.x = 0;
          sizes.y = 0;
          sizes.width = cropSize;
          sizes.height = cropSize;
  
          sizes.x = (texture.width - cropSize) / 2f;
          sizes.y = (texture.height - cropSize) / 2f;
  
  
          result.SetPixels(texture.GetPixels(Mathf.FloorToInt(sizes.x), Mathf.FloorToInt(sizes.y), Mathf.FloorToInt(sizes.width), Mathf.FloorToInt(sizes.height)));
          result.Apply();
          return result;
      }`

I hope that helps if anyone is looking for this. 
To use this, call PickImage(-1) and send the resulted texture to CropTexture(yourTextureName). This will return the cropped texture.

**Reply by yasirkula** [reply]
Congratulations on creating the first discussion in my repositories! If you'd like the user to crop their own images, you can have a look at [ImageCropper](https://github.com/yasirkula/UnityImageCropper). Also, I wouldn't recommend passing -1 to PickImage because the resulting Texture can be quite big and would consume a lot of memory.

**Reply by gokturk-tas** [reply]
Thanks for the feedback, I will take a look at ImageCropper and passing -1 issue.
